### ISSUE TITLE
Bug: Older TF version incompatible with the optimizer wrapper

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#32]
This seems to be related to `tf-addons` and not `GradientAccumulateOptimizer`. Hence, if we remove the `tf-addons` dependency, it should not be an issue. However, we use `tf-addons` for fancy type hints. Worth it?

This is the error log I get for `tensorflow==2.2.0`:

```
Run python -c "from gradient_accumulator import GradientAccumulateModel, GradientAccumulateOptimizer"
[7](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:8)
/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/tensorflow_addons/utils/ensure_tf_install.py:67: UserWarning: Tensorflow Addons supports using Python ops for all Tensorflow versions above or equal to 2.4.0 and strictly below 2.7.0 (nightly versions are not supported). 
[8](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:9)
 The versions of TensorFlow you are currently using is 2.2.0 and is not supported. 
[9](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:10)
Some things might work, some things might not.
[10](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:11)
If you were to encounter a bug, do not file an issue.
[11](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:12)
If you want to make sure you're using a tested and supported configuration, either change the TensorFlow version or the TensorFlow Addons's version. 
[12](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:13)
You can find the compatibility matrix in TensorFlow Addon's readme:
[13](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:14)
https://github.com/tensorflow/addons
[14](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:15)
  UserWarning,
[15](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:16)
Traceback (most recent call last):
[16](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:17)
  File "<string>", line 1, in <module>
[17](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:18)
  File "/home/runner/work/GradientAccumulator/GradientAccumulator/gradient_accumulator/__init__.py", line 1, in <module>
[18](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:19)
    from .accumulators import GradientAccumulateOptimizer
[19](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:20)
  File "/home/runner/work/GradientAccumulator/GradientAccumulator/gradient_accumulator/accumulators.py", line 3, in <module>
[20](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:21)
    from tensorflow_addons.utils import types
[21](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:22)
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/tensorflow_addons/__init__.py", line 21, in <module>
[22](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:23)
    from tensorflow_addons import activations
[23](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:24)
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/tensorflow_addons/activations/__init__.py", line 17, in <module>
[24](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:25)
    from tensorflow_addons.activations.gelu import gelu
[25](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:26)
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/tensorflow_addons/activations/gelu.py", line 19, in <module>
[26](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:27)
    from tensorflow_addons.utils.types import TensorLike
[27](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:28)
  File "/opt/hostedtoolcache/Python/3.6.15/x64/lib/python3.6/site-packages/tensorflow_addons/utils/types.py", line 26, in <module>
[28](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:29)
    from tensorflow.python.keras.engine import keras_tensor
[29](https://github.com/andreped/GradientAccumulator/actions/runs/4038830821/jobs/6943110335#step:7:30)
ImportError: cannot import name 'keras_tensor'
```

**Comment by andreped** [comment]
Then again, `tensorflow==2.3.0` works just fine. Perhaps we should just deprecate `tf==2.x`?

**Comment by andreped** [comment]
Fixed in 84f1893a48a3c292d8860d3a7892e3e5c4c735d6.

I ended up deprectating `tf==2.2`. Now we support `tf >= 2.3`.
